### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by davidzhangnyc**
WHAT DOES THIS PR DO:

Implemented robust testing using Enzyme

HOW TO TEST THIS PR:

gh pr checkout 43 && npm install && npm run test
verify that the tests all pass

### CODE REVIEW

**Review by JefZheng — APPROVED** [review]

**Review by Hydroelectric29 — APPROVED** [review]

**Review by JefZheng — APPROVED** [review]

File: __tests__/enzyme.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+/**
+ * @jest-environment jsdom
+ */
+
+
```
Comment (dlande000) [inline __tests__/enzyme.js]: Delete line 5.

File: __tests__/enzyme.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import React from 'react';
+import { configure, shallow, mount, render } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+
+// Enzyme is a wrapper around React test utilities which makes it easier to
+// shallow render and traverse the shallow rendered tree.
```
Comment (dlande000) [inline __tests__/enzyme.js]: Delete lines 10 and 11.

File: __tests__/enzyme.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
+/**
```
Comment (dlande000) [inline __tests__/enzyme.js:1]: Change indents from 4 spaces to 2 for entire file.

File: __tests__/enzyme.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+// Enzyme is a wrapper around React test utilities which makes it easier to
+// shallow render and traverse the shallow rendered tree.
+import NavBar from '../src/components/navBar';
+import BarChart from '../src/components/barChart';
+
+configure({ adapter: new Adapter() });
+
+describe('React unit tests', () => {
+    let wrapper;
+    describe('NavBar component', () => {
```
Comment (dlande000) [inline __tests__/enzyme.js]: Add blank line between 18 and 19.

File: __tests__/enzyme.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        const props = {
+            setIsLogShowing: jest.fn(() => console.log('setIsLogShowing')),
+            setAreMetricsShowing: jest.fn(() => console.log('setAreMetricsShowing')),
+            setIsAboutShowing: jest.fn(() => console.log('setIsAboutShowing')),
+        };
+        beforeAll(() => {
```
Comment (dlande000) [inline __tests__/enzyme.js]: add blank line between 24 and 25.

File: __tests__/enzyme.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+            wrapper = shallow(<NavBar {...props} />);
+        });
+
+        it('should render an unordered list of buttons', () => {
+            expect(wrapper.find('ul').length).toBe(1);
+        });
+            
+    });
+    describe('BarChart component', () => {
```
Comment (dlande000) [inline __tests__/enzyme.js]: Add blank line between 33 and 34.

File: __tests__/enzyme.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        let wrapper;
+        const props = {
```
Comment (dlande000) [inline __tests__/enzyme.js]: add blank line between 35 and 36.

File: __tests__/enzyme.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+            data: [],
+            categories: [],
+        }
+        beforeAll(() => {
```
Comment (dlande000) [inline __tests__/enzyme.js]: Add blank line between 39 and 40.

File: __tests__/enzyme.js [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+            wrapper = shallow(<BarChart {...props} />);
+        });
+
+        it('should render without errors', () => {
+            expect(wrapper.length).toBe(1);
+        });
+    });
+});
```
Comment (dlande000) [inline __tests__/enzyme.js]: Add blank line at end of file.

**Review by dlande000 — APPROVED** [review]
Approved pending minor style changes.

**Comment by dlande000** [comment]
To make sure your package-lock issue gets resolved correctly: 

1. `git pull origin testing`
2. Delete package-lock
3. Run `npm install`
4. `git checkout main && git pull upstream main && git checkout testing && git merge main`
5. Resolve all the merge conflicts in your package json by just accepting all incoming changes. 
6. `npm install && npm run dev` to verify the app still works
7. `npm run test` to verify that the tests still work.

**Pull request merged**